openverse_catalog/dags/providers/provider_api_scripts/wikimedia_commons.py

```diff
@@ -21,450 +21,428 @@
+                self.continue_token = response_json.pop("continue", {})
+                query_params.update(self.continue_token)
+                logger.info(f"new_continue_token: {self.continue_token}")
+
+                # Merge this response into the batch
+                batch_json = self.merge_response_jsons(batch_json, response_json)
+
+            if "batchcomplete" in response_json:
+                logger.info("Found batchcomplete")
+                break
+        return batch_json
+
+    def get_should_continue(self, response_json):
+        # Should not continue if continue_token is Falsy
+        return self.continue_token
+
+    def get_batch_data(self, response_json):
+        image_pages = self.get_media_pages(response_json)
+        if image_pages is not None:
+            return image_pages.values()
+        return None
 
-    funcs = {
-        IMAGE: _add_image,
-        AUDIO: _add_audio,
-    }
-    funcs[valid_mediatype](parsed_data, media_data, media_info)
-
-
-def _get_value_by_name(key_value_list: list, prop_name: str):
-    prop_list = [
-        key_value_pair
-        for key_value_pair in key_value_list
-        if key_value_pair["name"] == prop_name
-    ]
-    if prop_list:
-        return prop_list[0].get("value")
-
-
-def _get_value_by_names(key_value_list: list, prop_names: list):
-    """Gets the first available value for one of the `prop_names`
-    property names"""
-    for prop_name in prop_names:
-        if val := _get_value_by_name(key_value_list, prop_name):
-            return val
+    def get_media_pages(self, response_json):
+        if response_json is not None:
+            image_pages = response_json.get("query", {}).get("pages")
+            if image_pages is not None:
+                logger.info(f"Got {len(image_pages)} pages")
+                return image_pages
 
+        logger.warning(f"No pages in the image batch: {response_json}")
+        return None
 
-def _parse_audio_file_data(parsed_data: dict, file_metadata: list) -> dict:
-    streams = _get_value_by_name(file_metadata, "streams")
-    try:
+    def get_record_data(self, record):
+        foreign_id = record.get("pageid")
+        logger.debug(f"Processing page ID: {foreign_id}")
+
+        media_info = self.get_media_info_dict(record)
+
+        valid_media_type = self.extract_media_type(media_info)
+        if not valid_media_type:
+            # Do not process unsupported media types, like Video
+            return None
+
+        license_info = self.get_license_info(media_info)
+        if license_info.url is None:
+            return None
+
+        media_url = media_info.get("url")
+        if media_url is None:
+            return None
+
+        creator, creator_url = self.extract_creator_info(media_info)
+        title = self.extract_title(media_info)
+        filesize = media_info.get("size", 0)  # in bytes
+        filetype = self.extract_file_type(media_info)
+        meta_data = self.create_meta_data_dict(record)
```

Comment:
Could you open an issue for this, if you haven't already, @AetherUnbound ? It's nice to track it somewhere.

Reply:
Won't this be partially dependent on the class? `extract` and `get` have slightly different connotations to me. In this class `extract` felt useful because many of the utilities were a little more complex than simply pulling data out of a response.

If we really want to aim for consistency with/without `_` prefixing on the subclasses then I imagine we'd want a style guide or some sort of linting 🤔